src/Illuminate/Support/Str.php

```diff
@@ -779,4 +779,84 @@ public static function createUuidsNormally()
     {
         static::$uuidFactory = null;
     }
+
+    /**
+     * Clean a string with easy to remember options.
+     *
+     * @param $target
+     * @param array|null $options
+     *
+     * @return string
+     */
+    public static function clean($target, $options = null)
+    {
+        if (is_null($options)) {
+            $options = ['letters', 'numbers', 'space', 'comma', 'period', 'dash'];
+        }
+
+        $needles = [];
+
+        if (in_array('letters', $options) || in_array('alpha', $options)) {
+            $needles[] = 'a-zA-Z';
+        }
+
+        if (in_array('numbers', $options) || in_array('nums', $options)) {
+            $needles[] = '0-9';
+        }
+        if (in_array('comma', $options) || in_array(',', $options)) {
+            $needles[] = ',';
+        }
+        if (in_array('dash', $options) || in_array('-', $options) || in_array('hyphen', $options)) {
+            $needles[] = "\-";
+        }
+        if (in_array('dot', $options) || in_array('.', $options) || in_array('period', $options)) {
+            $needles[] = "\.";
+        }
+        if (in_array('colon', $options) || in_array(':', $options)) {
+            $needles[] = ':';
+        }
+        if (in_array('space', $options) || in_array(' ', $options)) {
+            $needles[] = ' ';
+        }
+
+        return preg_replace(sprintf('/[^%s]/', implode('', $needles)), '', $target);
+    }
+
+    /**
+     * Mask a string to a specific format, eg. Phone numbers, postal codes etc.
+     *
+     * @param string $target
+     * @param string $pattern
+     *
+     *  pattern should use the # character for numbers, and the $ character for letters
```

Comment:
invalid syntax. can't have floating text like this in between tags

Reply:
Ok - I think I'm getting the hang of it.  sorry Graham, First time - thanks for being patient with me.